File: motherless.js
const axios = require('axios');
const cheerio = require('cheerio');

module.exports = async function motherless(query) {
  const url = `https://motherless.com/videos?q=${encodeURIComponent(query)}`;
  const results = [];

  try {
    const { data } = await axios.get(url);
    const $ = cheerio.load(data);

    $('div.video-thumb').each((i, el) => {
      const title = $(el).find('.video-title').text().trim();
      const href = $(el).find('a').attr('href');
      const duration = $(el).find('.duration').text().trim();

      if (title && href) {
        results.push({
          title,
          url: 'https://motherless.com' + href,
          duration,
          source: "Motherless"
        });
      }
    });
  } catch (err) {
    console.error("motherless error:", err.message);
  }

  return results;
};

File: scrapers/motherless.py
import requests
from bs4 import BeautifulSoup

def scrape_motherless(query, mode="straight", page=1):
    results = []
    url = f"https://motherless.com/videos?q={query}&page={page}"
    headers = {'User-Agent': 'Mozilla/5.0'}
    r = requests.get(url, headers=headers, timeout=10)
    soup = BeautifulSoup(r.content, "html.parser")
    for vid in soup.select("div.thumbvideo-container"):
        try:
            a = vid.select_one("a.thumbvideo")
            if not a or not a.has_attr("href"):
                continue
            title = a.get('title') or a.text.strip()
            video_url = "https://motherless.com" + a['href']
            img = vid.select_one("img")
            preview = img.get("data-src") or img.get("src")
            results.append({
                "title": title,
                "url": video_url,
                "preview": preview,
                "source": f"Motherless"
            })
        except Exception:
            continue
    return results
